NguyenVanViet2004/FE-Shoes-Ecommerce
# Integrate useUserLocation, usePermissions into the project and fix some bugs

## Related issues
Resolved https://github.com/NguyenVanViet2004/FE-Shoes-Ecommerce/issues/44

## Summary
- Added `useUserLocation` custom hook to fetch user's location and address.
- Added `usePermissions` hook to manage location permissions.
- Fixed issue with navigation to `BottomBar` after scrolling through slides. Now correctly handles navigation based on the current slide index.


## How to test
1. **Permissions and Location**: 
   - Ensure the app requests location permissions and correctly handles denial.
   - Verify that the app fetches the user's current location and converts it to an address.
   - Check if the location and address are updated correctly in the `HomeTemplate` component.

2. **Navigation**:
   - Navigate through the slides in the `Onboarding`.
   - Confirm that the app navigates to the `BottomBar` screen when on the last slide.


## Concern points


## Screenshots
![Simulator Screenshot - iPhone 15 Pro - 2024-08-14 at 16 44 45](https://github.com/user-attachments/assets/e9110258-dcc9-4afe-ae5a-405ae9befe71)
